### PR TITLE
Ability to disable cluster / peers discovery

### PR DESCRIPTION
**Pull request by sgalsaleh**
I included some details in the README about how this works.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,16 +53,14 @@ conn, err := gorqlite.Open("https://mary:secret2@server1.example.com:4001/?level
...
+conn, err := gorqlite.Open("https://localhost:2265/?level=strong&timeout=30&disableClusterDiscovery=true")
 
 // change our minds
 conn.SetConsistencyLevel("none")
 conn.SetConsistencyLevel("weak")
 conn.SetConsistencyLevel("strong")
 
-// set the http timeout.  Note that rqlite has various internal timeouts, but this
-// timeout applies to the http.Client and its work.  It is measured in seconds.
-conn.SetTimeout(10)
```
Comment (sgalsaleh) [inline README.md:63]: This does not exist

File: conn.go [hunk 2]
```diff
@@ -113,6 +117,9 @@ func (conn *Connection) Leader() (string, error) {
 	if conn.hasBeenClosed {
 		return "", errClosed
 	}
+	if conn.disableClusterDiscovery {
```
Comment (otoolep) [inline conn.go:120]: Why this change? What is your thinking?

File: README.md [hunk 3]
```diff
@@ -52,16 +53,14 @@ conn, err := gorqlite.Open("https://mary:secret2@server1.example.com:4001/?level
 conn, err := gorliqte.Open("https://server2.example.com:4001/?level=weak")
 // different port, setting the rqlite consistency level and timeout
 conn, err := gorqlite.Open("https://localhost:2265/?level=strong&timeout=30")
+// different port, setting the rqlite consistency level, timeout, and disabling cluster discovery
+conn, err := gorqlite.Open("https://localhost:2265/?level=strong&timeout=30&disableClusterDiscovery=true")
```
Comment (otoolep) [inline README.md:57]: So the only thing about this is that it implies that `isableClusterDiscovery` is something that `rqlited` supports, which of course it doesn't. I understand why you did it -- and if this is the only way, that's what we will have to do -- but it could confuse people.

Instead, is it possible to add a new method to the `conn` object, something like `DisableClusterDiscovery(bool)`?

Comment (sgalsaleh) [reply README.md:57]: Good point. Yes, I can do that.

File: conn.go (hunk 2 above)
Comment (sgalsaleh) [reply conn.go:120]: This is so that updateClusterInfo isn't called, because if it is, we're back to the same problem. I don't mind if it happens since we're not using this function, but it feels wrong to have the behavior change just because this function got used. Same with Peers.

File: README.md (hunk 3 above)
Comment (sgalsaleh) [reply README.md:57]: Actually, I take that back. Parameters in the connection string shouldn't necessarily map to rqlite API options. Even `timeout` is not something that rqlited supports but is something is used to configure the client to time out after a certain period of time. Does that make sense?

Also, the cluster discovery happens when the connection is initialized, so it's going to be tricky to revert its changes later with something like `conn.DisableClusterDiscovery`.

Comment (otoolep) [reply README.md:57]: Yeah, your last point is the one I was afraid of (but just don't know the code well enough without looking at it).

**Comment by otoolep** [comment]
LGTM -- merge away.

**Comment by sgalsaleh** [comment]
Thank you. I just updated the README to make it clear that `disableClusterDiscovery` is a client configuration.

File: README.md (hunk 3 above)
Comment (sgalsaleh) [reply README.md]: We could make it a second parameter to the `Open` function, but unfortunately that would be a breaking change, and I'd like to avoid that.

Comment (otoolep) [reply README.md]: Yeah, agreed. This client library, even though it has plenty of rough edges, has been quite useful to folks and we owe it a lot. I think it needs to be rewritten (or a new one added to the repo which will be more inline with how rqlite works) but until that happens, we shouldn't break the API.

**Comment by otoolep** [comment]
All good ahead, after making sure the SSH key is ok.

https://app.circleci.com/pipelines/github/rqlite/gorqlite/82/workflows/0dae2025-53e4-4ed1-81bc-413c865649e8
